Jint.Tests/Runtime/IsolatedContextTests.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using FluentAssertions;
+using Xunit;
+
+namespace Jint.Tests.Runtime
+{
+    public class IsolatedContextTests
+    {
+        [Fact]
+        public void IsolatedContextCanBeUsedAndDisposed()
+        {
+            var engine = new Engine(options => options.Strict());
+            engine.SetValue("assert", new Action<bool>(Assert.True));
+
+            // Set outer variable in global scope
+            engine.SetValue("outer", 123);
+            engine.Execute("assert(outer === 123)");
+            engine.GetValue("outer").ToObject().Should().Be(123);
+
+            // Enter new execution context
+            using (engine.EnterIsolatedContext())
+            {
+                // Can see global scope
+                engine.Execute("assert(outer === 123)");
+
+                // Can modify global scope
+                engine.Execute("outer = 321");
+                engine.GetValue("outer").ToObject().Should().Be(321);
```

Comment:
What is the value when the scope is released?
Why doesn't it throw an exception like `Math.max`. Not saying it should, but how do we know it won't?

Reply:
I explained this below wrt `Math.max`, TLDR; we can rewrite and modify anything that can be attached to our isolated context - this is shadowing vars. `outer = 32;` is effectively `var outer = 321;` so introducing a new variable isolated environment hiding old from global.